meta-xt-control-domain/recipes-guest/doma/files/doma-restart-monitor-ExecStart.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+echo "DomA restart monitor started!";
+
+# Fetch DomA Xen domain identifier
+DOMA_ID="";
+
+while [ "$DOMA_ID" == "" ]; do
+    DOMA_ID=$(xl list | awk '{ if ($1 == "DomA") print $2 }');
+
+    if [ "$DOMA_ID" = "" ]; then
+        sleep 1
+    else
+        echo "Parsed DOMA_ID is '${DOMA_ID}'";
+    fi
+done
+
+XS_PATH="/local/domain/$DOMA_ID"
+
+# Fetch first availability of the parameter
+until xenstore-read $XS_PATH; do
```

Comment:
Hi @lorc, it will not work, as here I'm waiting for the availability of the root XS_PATH itself. It is not a leaf node of the xen-store. No one assigns any values to it, so xenstore-watch will stuck forever.

```
# Trying to work with non-running domain 3
# xenstore-watch -n 1 will return without errors independently from whether the path already exists or not. So it is not representative
# xenstore-watch -n 2 will stuck forever in case this service was started after the domain was already running. So also will not work to check when the domain has become available
root@generic-armv8-xt-dom0:~# xenstore-watch -n 1 /local/domain/3
/local/domain/3
# read operation will fail if there is no path existing. Can be used for the check. 
root@generic-armv8-xt-dom0:~# xenstore-read /local/domain/3                                                                                                                        
xenstore-read: couldn't read path /local/domain/3
```